src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
1. Again, we have exact same pieces of code like `people.filter((p) => p.sex === 'f')` 
2. `people.find((p1) => p1.mother === p.name))` has enough size to separate it into other function

Let's do:
`const isMotherHasChildren = name => people.find(({ mother }) => mother === name)); `
`const women = people.filter(({ sex, name }) => sex === 'f' && (!withChildren || isMotherHasChildren(name)))`